doc/source/cluster/kubernetes/user-guides/tpu.md

```diff
@@ -0,0 +1,71 @@
+
+## Configuring Ray Pods for TPU usage
+
+Using any TPU accelerator requires specifying `google.com/tpu` resource `limits` and `requests` in the container fields of your `RayCluster`'s
+`workerGroupSpecs`. This resource specifies the number of TPU chips for GKE to allocate each Pod. KubeRay v1.1.0 adds a `numOfHosts`
+field to the RayCluster custom resource, specifying the number of TPU hosts to create per worker group replica. For multi-host worker groups,
+replicas are treated as Pod slices rather than individual workers, with `numOfHosts` worker nodes being created per replica.
+Additionally, GKE uses `gke-tpu` node selectors to schedule TPU Pods on the node matching the desired TPU accelerator and topology.
+
+Below is a config snippet for a RayCluster worker group with 2 Ray TPU worker Pods, each scheduled on its own GKE v4 TPU node belonging to the same TPU Pod slice.
+
+```
+   groupName: tpu-group
+   replicas: 1
+   minReplicas: 0
+   maxReplicas: 1
+   numOfHosts: 2
+   ...
+   template:
+       spec:
+        ...
+        containers:
+         - name: ray-worker
+           image: rayproject/ray:2.9.0-py310
+           ...
+           resources:
+             google.com/tpu: "4" # Required to use TPUs.
+             ...
+           limits:
+             google.com/tpu: "4" # The resources and limits value is expected to be equal.
+             ...
+        nodeSelector:
+            cloud.google.com/gke-tpu-accelerator: tpu-v4-podslice
+            cloud.google.com/gke-tpu-topology: 2x2x2
+            ...
+```
+
+## TPU workload scheduling
+
+After a Ray pod with with TPU pod resources is deployed, it will be able to execute tasks and actors annotated with TPU requests.
+TPUs are supported on Ray as a [custom resource](https://docs.ray.io/en/latest/ray-core/scheduling/resources.html#custom-resources),
+and are requested by tasks or actors using the decorator `@ray.remote(resources={"TPU": NUM_TPUS})`.
+
+## Multi-Host TPU Autoscaling
+
+Multi-host TPU autoscaling is supported in Kuberay versions 1.1.0 or later and Ray versions 2.32.0 or later. Ray multi-host TPU worker groups are worker groups which specify "google.com/tpu" Kubernetes container limits/requests and have `NumOfHosts` greater than 1. Each replica of a Ray multi-host TPU worker group is treated as a TPU Pod slice and scaled atomically. When scaling up, multi-host worker groups create `NumOfHosts` Ray workers per replica. Likewise, multi-host worker group replicas are scaled down by `NumOfHosts` workers. When a single Ray worker in a multi-host TPU worker group is scheduled to be deleted, the entire replica to which it belongs is terminated. When scheduling TPU workloads on multi-host worker groups, it's important to ensure Ray tasks or actors run on every TPU VM host in a worker group replica to avoid idle TPU workers being scaled down.
```

Comment:
```suggestion
Multi-host TPU autoscaling is supported in Kuberay versions 1.1.0 or later and Ray versions 2.32.0 or later. Ray multi-host TPU worker groups are worker groups, which specify "google.com/tpu" Kubernetes container limits or requests and have `NumOfHosts` greater than 1. Ray treats each replica of a Ray multi-host TPU worker group as a TPU Pod slice and scales them atomically. When scaling up, multi-host worker groups create `NumOfHosts` Ray workers per replica. Likewise, Ray scales down multi-host worker group replicas by `NumOfHosts` workers. When a Ray schedules to delete a single Ray worker in a multi-host TPU worker group, Ray terminates the entire replica to which the worker belongs. When scheduling TPU workloads on multi-host worker groups, it's important to ensure Ray tasks or actors run on every TPU VM host in a worker group replica to avoid Ray from scaling down idle TPU workers.
```